tools/travis/before_install.sh

```diff
@@ -29,8 +29,12 @@ export WHEELHOUSE
 export DISPLAY=:99.0
 # This causes way too many internal warnings within python.
 # export PYTHONWARNINGS="d,all:::skimage"
-# codecov will combine different paths
-export TEST_ARGS="--doctest-modules --cov=skimage"
+if [[ "$OPTIONAL_DEPS" == "1" ]]; then
+  # We must run doctests with the optional deps since
+  # it seems to crawl through the code base. if it cannot find
+  # matplotlib, it causes an error when it gathers the tests
+  export TEST_ARGS="--doctest-modules --cov=skimage"
```

Comment:
Just bringing attention to this specific line. I really don't know how to deal with these doctests. @stefanv brought this up earlier in https://github.com/scikit-image/scikit-image/pull/3311#discussion_r231336994

Reply:
I had to add a test dependenecy on doctestplus https://github.com/astropy/pytest-doctestplus/